File: sdks/php/src/Dialect/MySQLDialect.php
<?php
namespace Autonoma\Sdk\Dialect;

class MySQLDialect implements DialectInterface
{
    public function getName(): string { return 'mysql'; }
    public function supportsReturning(): bool { return false; }
    public function param(int $_index): string { return '?'; }
    public function quoteId(string $name): string { return '`' . $name . '`'; }
    public function tablesSql(string $schema): string { return SqlQueries::replaceSchema(SqlQueries::MYSQL_TABLES, $schema); }
    public function columnsSql(string $schema): string { return SqlQueries::replaceSchema(SqlQueries::MYSQL_COLUMNS, $schema); }
    public function primaryKeysSql(string $schema): string { return SqlQueries::replaceSchema(SqlQueries::MYSQL_PRIMARY_KEYS, $schema); }
    public function foreignKeysSql(string $schema): string { return SqlQueries::replaceSchema(SqlQueries::MYSQL_FOREIGN_KEYS, $schema); }
    public function enumsSql(string $schema): string { return SqlQueries::MYSQL_ENUMS; }
}

File: sdks/java/autonoma-sdk/src/main/java/ai/autonoma/sdk/MysqlDialect.java
package ai.autonoma.sdk;

public class MysqlDialect implements Dialect {

    @Override public String name() { return "mysql"; }
    @Override public String param(int index) { return "?"; }
    @Override public String quoteId(String name) { return "`" + name + "`"; }
    @Override public boolean supportsReturning() { return false; }

    @Override
    public String tablesSQL(String schema) {
        return SqlQueries.MYSQL_TABLES.replace("{{schema}}", schema);
    }

    @Override
    public String columnsSQL(String schema) {
        return SqlQueries.MYSQL_COLUMNS.replace("{{schema}}", schema);
    }

    @Override
    public String primaryKeysSQL(String schema) {
        return SqlQueries.MYSQL_PRIMARY_KEYS.replace("{{schema}}", schema);
    }

    @Override
    public String foreignKeysSQL(String schema) {
        return SqlQueries.MYSQL_FOREIGN_KEYS.replace("{{schema}}", schema);
    }

    @Override
    public String enumsSQL(String schema) {
        return SqlQueries.MYSQL_ENUMS;
    }
}
